macros/src/main/scala/com/timeout/CodeGen.scala

```diff
@@ -64,14 +75,39 @@ class CodeGen(conf: CodeGen.Config) {
     SpecsParser.resourceTypes(conf.excludePrefixes).map { rt =>
       val normalized = normalize(rt.fqn)
       val typeName = Type.Name(normalized)
-
       val namespace = normalize(rt.fqn.takeWhile(_ != "."))
+
       val logicalIdProp: Term.Param = param"logicalId: String"
-      val properties = logicalIdProp :: rt.properties.map(mkField(_, Some(namespace)))
+      val dependsOn: Term.Param = param"DependsOn: Option[CfExp[String]] = None"
+      val deletionPolicy: Term.Param = param"DeletionPolicy: Option[CfExp[DeletionPolicy]] = None"
+      val creationPolicy: Term.Param = param"CreationPolicy: Option[CfExp[CreationPolicy]] = None"
+      val updatePolicy: Term.Param = param"UpdatePolicy: Option[CfExp[List[UpdatePolicy]]] = None"
+
+      val commonProps_ =
+        logicalIdProp :: dependsOn :: creationPolicy :: deletionPolicy :: Nil
+
+      val commonProps = if (rt.fqn == "AWS::AutoScaling::AutoScalingGroup")
+        commonProps_ ++ List(updatePolicy)
+      else
+        commonProps_
+
+      val properties = commonProps ++ rt.properties.map(mkField(_, Some(namespace)))
       val fqn = Term.Name("\"" + rt.fqn + "\"")
 
-      val declaration = q"""case class $typeName (..$properties) extends Resource {
-            override def fqn: String = $fqn
+      val jsonFields = rt.properties.map { prop =>
+        q"${Lit(prop.name)} -> ${Term.Name(prop.name)}.asJson"
+      }
+
+      val declaration = q"""
+         case class $typeName (..$properties) extends Resource {
+           override def fqn: String = $fqn
+           override def jsonEncode: io.circe.Json =
+             Json.obj(
+               logicalId -> Json.obj(
+                 "Type" -> Json.fromString(fqn),
+                 "Properties" -> Json.obj(..$jsonFields)
+               )
```

Comment:
Actually, `DependsOn`, `DeletionPolicy`, `CreationPolicy`, and `UpdatePolicy`, go in here!